src/sensors/rev_encoder.hpp

```diff
@@ -1,16 +1,58 @@
+/**
+ * @file rev_encoder.hpp
+ * @brief The header file for the Rev Encoder firmware
+ * @date 2023-10-31
+ */
+
+#include <cmath>
 #include <FreqMeasureMulti.h>
 
 #ifndef REV_ENCODER_H
 #define REV_ENCODER_H
 
-class RevEncoder {
+/**
+ * @brief the class for the Rev Encoder
+ */
+class RevEncoder 
+{
 	private:
-		uint8_t pin;
+		/**
+		* @brief the pin number that the encoders signal pin is plugged into
+		*/
+		uint8_t inPin;
+		/**
+		* @brief Used to read rise time of the encoder
+		*/
 		FreqMeasureMulti freq;
+		/**
+		* @brief measure of current angle in ticks (1024 ticks = 1 rotation). When set to 1 individual readings are returned
+		*/
+		float ticks;
+		/**
+		* @brief measure of current angle in radians 
+		*/
+		float radians;
 	public:
-		RevEncoder(uint8_t pin);
-		int getRawOutput();
-		float getAngle();
+		/**
+		* @brief Construct a new rev_encoder object
+		* @param encPin the pin number that the encoders signal pin is plugged into
+		* @param baudrate the desired baudrate for the encoder
+		*/
+		RevEncoder(uint8_t encPin, int baudrate);
+		/**
+		* @brief updates the current angle in ticks (1024 = 1 Rotation) and radians.
```

Comment:
What is the difference between ticks and radians? If there is no difference, do we need both?

Also, remember to switch to /// (we do this so we can more independently change lines of doxygen comments, and it makes it explicit what is doxygen vs what is not.

Reply:
it's up to rishi if he doesn't want both I can remove one.

Reply:
Do they measure the same thing? If they do, it would be better to remove one (or if you use the less-intuitive one internally, keep it private, while keeping the intuitive output getter public)